app/src/test/java/com/codesoom/assignment/controller/ProductReadControllerTest.java

```diff
@@ -0,0 +1,100 @@
+package com.codesoom.assignment.controller;
+
+import com.codesoom.assignment.application.ProductNotFoundException;
+import com.codesoom.assignment.application.ProductReadServiceImpl;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.Assertions.*;
+
+
+@ActiveProfiles("test")
+@SpringBootTest
+public class ProductReadControllerTest {
+
+    private ProductReadController controller;
+
+    @Autowired
+    private ProductReadServiceImpl service;
+
+    @Autowired
+    private ProductRepository repository;
+
+    private static final Product SAVED_PRODUCT
+            = new Product("쥐돌이", "캣이즈락스타", BigDecimal.valueOf(4000), "");
+
+    @BeforeEach
+    void setup() {
+        this.controller = new ProductReadController(service);
+        repository.deleteAll();
+    }
```

Comment:
지금 당장 이렇게까지 할 필요는 없지만, 이렇게 할 수 있다는 걸 기억해두시면 언젠간 도움이 될 거에요.
```suggestion
    @BeforeEach
    void setup() {
        this.controller = new ProductReadController(service);
        clear();
    }

    @AfterEach
    void clear() {
        repository.deleteAll();
    }

```